python/lint/action.yaml

```diff
@@ -36,6 +36,7 @@ runs:
       with:
         options: --check --config ${{ github.action_path }}/pyproject.toml
         src: "."
+        version: 23.12.1
```

Comment:
Imo it's not a way to go. 

Either we could update [pyproject.toml](https://github.com/PiwikPRO/actions/blob/pin-black-to-23.12.1/python/lint/pyproject.toml) and update changed default to preserve compatibility or add an input argument that would allow teams to pin specific version if they really need to.

Reply:
Why don't we just apply to those new defaults instead?

Reply:
now flake8 fails.. I'm merging this PR for now, and after discussion on samurais we'll see what to do